cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -719,6 +781,78 @@ private void deleteKubeNamespace(String namespace, String cdapNamespace) throws
+   */
+  private void findOrCreateWorkloadIdentityConfigMap(String k8sNamespace) throws ApiException, IOException {
+    // Check if workload identity config map already exists
+    try {
+      coreV1Api.readNamespacedConfigMap(k8sNamespace, WORKLOAD_IDENTITY_CONFIGMAP_NAME, null, false, false);
+      // Workload identity config map already exists, so return early
+      LOG.debug("Workload identity config found, returning without creating it...");
+      return;
+    } catch (ApiException e) {
+      if (e.getCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+        LOG.debug("Creating workload identity config map for kubernetes namespace {}", k8sNamespace);
+      } else {
+        throw new IOException("Failed to fetch existing workload identity config map. Error code = " + e.getCode() +
+          ", Body = " + e.getResponseBody(), e);
+      }
+    }
+
+    String workloadIdentityAudience = String.format(WORKLOAD_IDENTITY_AUDIENCE_FORMAT, workloadIdentityPool,
+                                                    workloadIdentityProvider);
+    String workloadIdentityImpersonationURL = String.format(WORKLOAD_IDENTITY_IMPERSONATION_URL_FORMAT,
+                                                            workloadIdentityGCPServiceAccountEmail);
+    GCPWorkloadIdentityCredential credential =
+      new GCPWorkloadIdentityCredential(GCPWorkloadIdentityCredential.CredentialType.EXTERNAL_ACCOUNT,
+                                        workloadIdentityAudience, workloadIdentityImpersonationURL,
+                                        GCPWorkloadIdentityCredential.TokenType.JWT, WORKLOAD_IDENTITY_TOKEN_URL,
+                                        WORKLOAD_IDENTITY_CREDENTIAL_KSA_SOURCE_PATH);
+    String workloadIdentityCredentialJSON = gson.toJson(credential);
+    Map<String, String> workloadIdentityConfigMapData = new HashMap<>();
+    workloadIdentityConfigMapData.put(WORKLOAD_IDENTITY_DATA_KEY, workloadIdentityCredentialJSON);
+    V1ConfigMap configMap = new V1ConfigMap()
+      .metadata(new V1ObjectMeta().namespace(k8sNamespace).name(WORKLOAD_IDENTITY_CONFIGMAP_NAME))
+      .data(workloadIdentityConfigMapData);
+    coreV1Api.createNamespacedConfigMap(k8sNamespace, configMap, null, null, null);
+  }
+
+  /**
+   * Applies workload identity configurations to a given pod specification. For additional details, see steps 6-7 of
+   * https://cloud.google.com/anthos/multicluster-management/fleets/workload-identity#impersonate_a_service_account
+   * @param podSpec The pod spec to setup workload identity for.
+   */
+  private static void setupWorkloadIdentityForPodSpec(V1PodSpec podSpec, String workloadIdentityPool) {
+    // Mount volume to expected directory
+    V1ServiceAccountTokenProjection serviceAccountTokenProjection = new V1ServiceAccountTokenProjection()
+      .path(WORKLOAD_IDENTITY_CREDENTIAL_KSA_PATH)
+      .expirationSeconds(172800L)
```

Comment:
Can we have a config or at least a constant for this 2 days value?

Reply:
Switched to config, thanks!